src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.hotspot/src/jdk/vm/ci/hotspot/HotSpotResolvedJavaMethodImpl.java

```diff
@@ -577,6 +577,9 @@ public Type[] getGenericParameterTypes() {
 
     @Override
     public boolean canBeInlined() {
+        if (isForceInline()) {
+            return true;
+        }
```

Comment:
Should this use shouldBeInlined() instead?  Would it make sense for shouldBeInlined() to return true but canBeInlined() returns false?

Reply:
Yes as `shouldBeInlined` boils down to calling `CompilerOracle::should_inline` which can return true even if `Method::is_not_compilable` returns false.